presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformExistsApplyToLateralNode.java

```diff
@@ -38,20 +43,35 @@
 import java.util.Optional;
 
 import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
 import static com.facebook.presto.sql.planner.plan.LateralJoinNode.Type.INNER;
+import static com.facebook.presto.sql.planner.plan.LateralJoinNode.Type.LEFT;
 import static com.facebook.presto.sql.planner.plan.Patterns.applyNode;
+import static com.facebook.presto.sql.tree.BooleanLiteral.TRUE_LITERAL;
 import static com.facebook.presto.sql.tree.ComparisonExpressionType.GREATER_THAN;
+import static com.google.common.base.Preconditions.checkState;
 import static com.google.common.collect.Iterables.getOnlyElement;
 import static java.util.Objects.requireNonNull;
 
 /**
- * Exists is modeled as:
+ * EXISTS is modeled as (if correlated predicates are equality comparisons):
```

Comment:
The idea is:
```
- LIMIT 1
  - source[output:=(a,b,c)]
```
can be rewritten to:
```
- AggregationNode[groupingKeys:=(a,b,c)]
  - source[output:=(a,b,c)]
``` 
if we know that columns `(a,b,c)` are essentially a constant values.

In subquery context `(a,b,c)` will be "constant" values if they are part of filter predicates like:
```
a=correlated_column_x
b=correlated_column_y
c=correlated_column_z
```
e.g:
```
FilterNode[a=correlated_column_x, b=correlated_column_y, c=correlated_column_z]
```

Reply:
Without `(a,b,c)` being "constants" one cannot rewrite `LIMIT 1` to `AggregationNode[groupingKeys:=(a,b,c)]`, because there might be more the one group.